lib/common-types.ts

```diff
@@ -1668,13 +1668,13 @@ export type MRActions = {
     update: { params: GetUIConfigParams; payload: UIConfigProps; return: UIConfigProps }
   }
   Upload: {
-    get: { params: GetSpaceParams & { uploadId: string }; return: any }
+    get: { params: GetSpaceEnvironmentParams & { uploadId: string }; return: any }
```

Comment:
`GetSpaceEnvironmentParams & { uploadId: string }`

we repeat this string everywhere and it represents the fully upload params. why not give it a named type instead of repeating the union type everywhere?

eg `UploadParams`

Reply:
Done -  created this instead 
```
export type GetSpaceEnvironmentUploadParams = {
  spaceId: string
  environmentId: string
  uploadId?: string
}
```

Reply:
We don't want the required ` & {uploadId: string}}` here anymore then, right?

Reply:
I think we do because for gets and deletes, the uploadId is required. I only made this type for the generalized `getUploadUrlPath` because that creates the path for post/get/delete and the post doesn't need the uploadId but get/delete does so I made it optional to make the function work for all three http action types

Reply:
Per my comment we can remove the optional uploadId and bring this change back in (ie to use the main `GetSpaceEnvironmentUploadParams` type)